### DISCUSSION TITLE
Heap profile is not implemented in current build

### DISCUSSION THREAD

**Discussion by Freyert** [istio/istio#53103] [Q&A] [answered]
I wanted to make use of the `http://127.0.0.1:15000/heap_dump` endpoint. For some reason when I pull the image `docker.io/istio/proxyv2:1.23.1` in my local `kind` cluster I can use the endpoint, but when my GKE cluster pulls this same image tag I receive the text: `"Heap profile is not implemented in current build"`.

I see others have run into this, but they didn't seem to find a workaround: https://github.com/istio/istio/issues/48023#issuecomment-1827258824

**Comment by howardjohn** [comment, marked as answer]
I don't see how that can be the case with `docker.io/istio/proxyv2:1.23.1`. Are you 100% sure its that image exactly? Verified on my cluster it works fine. And it looks like that string `"Heap profile is not implemented in current build"` doesn't exist in the binary at all

**Reply by Freyert** [reply]
Ack, you're right.

Kind cluster is using `docker.io/istio/proxyv2:1.23.1` while GKE is using `docker.io/istio/proxyv2:1.23.3`

**Reply by Freyert** [reply]
Need to follow directions for 1.21 :)
